addons/http_routing/models/ir_http.py

```diff
@@ -144,7 +146,7 @@ def url_for(path_or_uri, lang=None):
 
         langs = request.env['ir.http']._get_languages()
         lang = pycompat.to_text(lang or request.context.get('lang', 'en_US'))
-        lang_url_code = langs.filtered(lambda l: l.code == lang or l.url_code == lang).url_code
+        lang_url_code = langs.filtered(lambda l: l.code == lang or l.url_code == lang).url_code or lang
```

Comment:
lang and langs.filter(lang)  or (request.lang in langs  and request.lang)

Reply:
Done something similar